incubator/hnc/test/conversion/conversion_test.go

```diff
@@ -79,6 +72,53 @@ metadata:
 		// Verify subnamespace anchor status in the new version.
 		FieldShouldContainWithTimeout(anchorCRD, nsA, nsB, ".apiVersion", "v1alpha2", crdConversionTime)
 		FieldShouldContain(anchorCRD, nsA, nsB, ".status.status", "Ok")
+		FieldShouldContain("ns", "", nsB, ".metadata.annotations", "map[hnc.x-k8s.io/subnamespace-of:"+nsA+"]")
+	})
+
+	It("should respect the old 'subnamespaceOf' annotation AT conversion if both exists", func() {
+		// Before conversion, create namespace A and a subnamespace B.
+		createSampleV1alpha1Subnamespace()
+		// Add `subnamespace-of` annotation, which should be unknown to HNC v0.5.
+		MustRun("kubectl annotate ns", nsB, "hnc.x-k8s.io/subnamespace-of=wrongvalue")
+		// Both subnamespace annotations should be there.
+		FieldShouldContain("ns", "", nsB, ".metadata.annotations", "hnc.x-k8s.io/subnamespaceOf:"+nsA)
+		FieldShouldContain("ns", "", nsB, ".metadata.annotations", "hnc.x-k8s.io/subnamespace-of:wrongvalue")
```

Comment:
Hmm, I'm not sure if I agree with this. It seems to make the logic more complicated but for not much gain. We could just say "if the new one exists, it _always_ overrules the old one" because I can't really imagine any reasonable cases where it would accidentally have the wrong value.

Reply:
sgtm.